sdk/nodejs/index.ts

```diff
@@ -0,0 +1,489 @@
+ *
+ * - chunklist(["id1", "id2", "id3"], 1) returns [["id1"], ["id2"], ["id3"]]
+ * - chunklist(["id1", "id2", "id3"], 2) returns [["id1", "id2"], ["id3"]]
+ *
+ * @param input the list to chunk
+ * @param size the size of each chunk
+ */
+export function chunklist<T>(input: T[], size: number): T[][] {
+    const temp = input.slice(0);
+    const results = [];
+
+    while (temp.length) {
+        results.push(temp.splice(0, size));
+    }
+    return results;
+}
+
+/**
+ * coalesce returns the first non-empty string from the given arguments, or an empty string if all
+ * of the arguments are undefined.
+ *
+ * @param first a potentially non-empty stringj
+ * @param others other potentially non-empty values
+ */
+export function coalesce(first: string | undefined, ...others: (string | undefined)[]): string {
+    if (first !== undefined && first !== "") {
+        return first;
+    }
+
+    for (const entry of others) {
+        if (entry !== undefined && entry !== "") {
+            return entry;
+        }
+    }
+
+    return "";
+}
+
+/**
+ * coalescelist returns the first non-empty list from the given arguments, or an empty list if
+ * all of the arguments are undefined or empty lists.
+ *
+ * @param first a potentially non-empty array
+ * @param others other potentially non-empty arrays
+ */
+export function coalescelist<T>(first: T[] | undefined, ...others: (T[] | undefined)[]): T[] {
+    if (first !== undefined && first.length !== 0) {
+        return first;
+    }
+
+    for (const entry of others) {
+        if (entry !== undefined && entry.length !== 0) {
+            return entry;
+        }
+    }
+
+    return [];
+}
+
+/**
+ * compact removes empty string elements from a list. This can be useful in some cases,
+ * for example when passing joined lists as module variables or when parsing module outputs.
+ *
+ * @param input a string array from which to remove empty values
+ */
+export function compact(input: string[]): string[] {
+    return input.filter(x => x !== "");
+}
+
+/**
+ * concat combines two or more lists into a single list.
+ *
+ * @param first the first list
+ * @param others other lists
+ */
+export function concat<T>(first: T[], ...others: T[][]): T[] {
+    let result = first;
+    for (const entry of others) {
+        result = result.concat(entry);
+    }
+    return result;
+}
+
+/**
+ * contains returns true if an array contains the given element and returns false otherwise
+ * @param haystack the array in which to search
+ * @param needle the element for which to search
+ */
+export function contains<T>(haystack: T[], needle: T): boolean {
+    return haystack.indexOf(needle) > -1;
+}
+
+export function dirname(path: string): string {
+    return Path.win32.dirname(path);
+}
+
+export function distinct(input: string[]): string[] {
+    return input.filter((elem, idx) => input.indexOf(elem) === idx);
+}
+
+export function element(inputList: string[], elementIndex: number) {
+    if (list.length === 0) {
```

Comment:
Should this be `inputList`?  Not actually sure how this is legal TS, I guess there is a global `list` somewhere in builtins?

Reply:
Yes, not sure how that got past the compiler. Must be something named `list` in scope.